queries/places.py

```diff
@@ -0,0 +1,352 @@
+
+    # Use top result
+    place = res["candidates"][0]
+
+    # Make sure it's in Iceland
+    coords = _parse_coords(place)
+    if not coords or not in_iceland(coords):
+        return gen_answer(_NOT_ICELAND_ERRMSG)
+
+    # Remove superfluous "Ísland" in addr string
+    addr = re.sub(r", Ísland$", "", place["formatted_address"])
+    # Get street name without number to get preposition
+    street_name = addr.split()[0].rstrip(",")
+    prep = iceprep_for_street(street_name)
+    # Split addr into street name w. number, and remainder
+    street_addr = addr.split(",")[0]
+    remaining = re.sub(r"^{0}".format(street_addr), "", addr)
+    # Get street name in dative case
+    addr_þgf = NounPhrase(street_addr).dative
+    # Assemble final address
+    final_addr = "{0}{1}".format(addr_þgf, remaining)
+
+    # Create answer
+    answer = final_addr
+    voice = "{0} er {1} {2}".format(placename, prep, numbers_to_neutral(final_addr))
+    response = dict(answer=answer)
+
+    return response, answer, voice
+
+
+def answ_openhours(placename, loc, qtype):
+    # Look up placename in places API
+    res = query_places_api(
+        placename,
+        userloc=loc,
+        fields="opening_hours,place_id,formatted_address,geometry",
+    )
+    if res["status"] != "OK" or "candidates" not in res or not res["candidates"]:
+        return gen_answer(_PLACES_API_ERRMSG)
+
+    from pprint import pprint
+    pprint(res)
+
+    # Use top result
+    place = res["candidates"][0]
+    if not "opening_hours" in place:
+        return gen_answer("Ekki tókst að sækja opnunartíma fyrir " + icequote(placename))
+
+    place_id = place["place_id"]
+    is_open = place["opening_hours"]["open_now"]
+    # needs_disambig = len(res["candidates"]) > 1
+    fmt_addr = place["formatted_address"]
+
+    # Make sure it's in Iceland
+    coords = _parse_coords(place)
+    if not coords or not in_iceland(coords):
+        return gen_answer(_NOT_ICELAND_ERRMSG)
+
+    # Look up place ID in Place Details API to get more information
+    res = query_place_details(place_id, fields="opening_hours,name")
+    if res["status"] != "OK" or not res or "result" not in res:
+        return gen_answer(_PLACES_API_ERRMSG)
+
+    pprint(res)
+
+    now = datetime.utcnow()
+    # Sun is index 0, as req. by Google API
+    wday = int(now.strftime("%w"))
+
+    try:
+        name = res["result"]["name"]
+        # Generate placename w. street, e.g. "Forréttabarinn á Nýlendugötu"
+        street = fmt_addr.split()[0].rstrip(",")
+        street_þgf = "{nl:þgf}".format(nl=NounPhrase(street))
+        name = "{0} {1} {2}".format(name, iceprep_for_street(street), street_þgf)
+
+        # Get opening hours for current weekday
+        periods = res["result"]["opening_hours"]["periods"]
+        if len(periods) == 1 or wday > len(periods)-1:
+            # Open 24 hours a day
+            today_desc = p_desc = "{0} er opin allan sólarhringinn".format(name)
```

Comment:
You could try casting the name into a NounPhrase and asking for its gender, before deciding to use `"opinn/opið/opin"` (with neutral gender as a default, I guess).